### PR TITLE
Optimize C# symbol content string allocation

### PR DESCRIPTION
**Pull request by Yves57**
See #873 
Follows the optimization #908 done for the HTML symbols.

### CODE REVIEW

**Comment by Yves57** [comment]
Continuous Integration problems seems to be out my my PR (seems to be here before my PR). Do you agree?

File: src/Microsoft.AspNetCore.Razor.Evolution/Legacy/CSharpTokenizer.cs [hunk 1]
```diff
@@ -168,6 +168,181 @@ protected override StateResult Dispatch()
...
+                    case CSharpSymbolType.Increment:
+                        return "++";
+                    case CSharpSymbolType.LessThanEqual:
+                        return "<=";
+                    case CSharpSymbolType.LeftShift:
+                        return "<<";
+                    case CSharpSymbolType.Equals:
+                        return "==";
+                    case CSharpSymbolType.GreaterThanEqual:
+                        if (Buffer[0] == '=')
```
Comment (NTaylorMullen) [inline src/Microsoft.AspNetCore.Razor.Evolution/Legacy/CSharpTokenizer.cs]: Why do we need to check this here? Does the tokenizer also treat `=>` as Greater Than Equal?

**Comment by NTaylorMullen** [comment]
> Continuous Integration problems seems to be out my my PR (seems to be here before my PR). Do you agree?

Possible. It's on my plate to investigate that today so we'll avoid merging this until those are resolved whether this causes it or not.

File: src/Microsoft.AspNetCore.Razor.Evolution/Legacy/CSharpTokenizer.cs (hunk 1 above)
Comment (Yves57) [inline src/Microsoft.AspNetCore.Razor.Evolution/Legacy/CSharpTokenizer.cs]: [Yes](https://github.com/aspnet/Razor/blob/dev/src/Microsoft.AspNetCore.Razor.Evolution/Legacy/CSharpTokenizer.cs#L111).
I don't know why it is the same token type but...

Comment (NTaylorMullen) [inline src/Microsoft.AspNetCore.Razor.Evolution/Legacy/CSharpTokenizer.cs]: 🤦‍♂️

**Comment by rynowak** [comment]
@NTaylorMullen is this good to go?

**Comment by NTaylorMullen** [comment]
> @NTaylorMullen is this good to go?

Yup, will merge it

**Comment by NTaylorMullen** [comment]
Merged. 5fa4672c99d9c7ce6ecb6ed72e0c841cc8287095

**Pull request closed without merging**